common/ptr.h

```diff
@@ -349,15 +387,25 @@ class WeakPtr : public BasePtr<T> {
 	WeakPtr(std::nullptr_t) : BasePtr<T>() {
 	}
 
-	template<class T2>
-	explicit WeakPtr(T2 *p) : BasePtr<T>(p) {
+	WeakPtr(const WeakPtr<T> &r) : BasePtr<T>(r) {
```

Comment:
I think this one is covered by the one below.

Reply:
No, that one is required and there were some nasty bugs without it.  If there is no copy constructor (even if another constructor could theoretically be a valid overload), then the compiler will generate a copy constructor, but the generated one will only copy the members and won't increment the reference count, causing a refcount desync and eventually access of an invalid tracker pointer.